src/hooks/keyword_hook.ts

```diff
@@ -0,0 +1,49 @@
+import fs from "fs"
+
+// 디렉토리에서 파일의 목록을 가져오는 함수, 키워드 체크용
+function getFilesInDirectory(path: string): Promise<string[]> {
+  return new Promise<string[]>((resolve, reject) => {
+    fs.readdir(path, (error, files) => {
+      if (error) {
+        reject(error);
+      } else {
+        resolve(files);
+      }
+    });
+  });
+}
+
+// 위 함수 핸들러?
+async function readDirHandler(path: string): Promise<string[]> {
+	try {
+		const contents: string[] = await getFilesInDirectory(path)
+		return contents
+	} catch(err) {
+		console.error('Faile to read File', err)
+		return []
+	}	
+}
+
+//  키워드를 받아서 "42docs/keyword.txt"로 파일을 참조하여 내용을 반환하는 함수
+async function readFileAsync(keyword: string): Promise<string> {
+	const path: string = './42docs/'.concat(keyword, '.txt')
+	return new Promise((resolve, reject) => {
+		fs.readFile(path, 'utf8', (err, data)=>{
+			if (err) {
+				reject(err)
+			} else {
+				resolve(data)
+			}
+		})
+	})
+}
+
+// 위 함수 핸들러
+async function loadShortDesc(keyword: string): Promise<string> {
+	try {
+		const contents = await readFileAsync(keyword)
+		return contents
+	} catch(err) {
+		return 'Not Found Keyword!!'
+	}	
+}
```

Comment:
[promises api](https://nodejs.org/api/fs.html#promises-api)와 [path api](https://nodejs.org/api/path.html#pathjoinpaths)를 사용하면 조금 더 읽기 좋은 코드가 될 수 있을 것 같아요!

```suggestion
import { promises } from "fs"
import path from "path"

/** "42docs/{키워드}.{확장자}" 경로의 내용을 읽는 함수
 *
 * @param extension 파일 확장자
 * @returns 텍스트 파일 내용
 */
const readKeywordFile = async (
	keyword: string,
	extension = ".txt",
): Promise<string> => {
	const filePath = path.join(__dirname, "42docs", keyword, extension)

	return promises.readFile(filePath, "utf8")
}

/** 키워드의 짧은 설명을 불러오는 함수 */
export const loadShortDesc = async (keyword: string): Promise<string> => {
	try {
		return readKeywordFile(keyword)
	} catch {
		return "키워드를 찾을 수 없습니다!"
	}
}
```